src/main/java/net/mcreator/ars_technica/common/blocks/SourceEngineBlockEntity.java

```diff
@@ -228,21 +225,21 @@ protected Vec3 getSouthLocation() {
         }
 
         @Override
-        public Vec3 getLocalOffset(BlockState state) {
+        public Vec3 getLocalOffset(LevelAccessor level, BlockPos pos, BlockState state) {
             Direction facing = state.getValue(SourceEngineBlock.FACING);
-            return super.getLocalOffset(state).add(Vec3.atLowerCornerOf(facing.getNormal())
+            return super.getLocalOffset(level,pos,state).add(Vec3.atLowerCornerOf(facing.getNormal())
                     .scale(-1 / 16f));
         }
 
         @Override
-        public void rotate(BlockState state, PoseStack ms) {
-            super.rotate(state, ms);
+        public void rotate(LevelAccessor level, BlockPos pos, BlockState state, PoseStack ms) {
+            super.rotate(level, pos, state, ms);
             Direction facing = state.getValue(SourceEngineBlock.FACING);
             if (facing.getAxis() == Direction.Axis.Y)
                 return;
             if (getSide() != Direction.UP)
                 return;
-            TransformStack.cast(ms)
+            TransformStack.of(ms)
                     .rotateZ(-AngleHelper.horizontalAngle(facing) + 180);
```

Comment:
I recall having some issue with this one when migrating to Create 6.0.0 for 1.21 hmmm... but I don't fully remember.
```suggestion
            TransformStack.of(ms)
                    .rotateZDegrees(-AngleHelper.horizontalAngle(facing) + 180);
```

Reply:
I’ve seen a lot of this code when mods were migrating to Create 6+. There’s no problem happening yet, but who knows.